modules/mobianRtdProvider.js

```diff
@@ -70,6 +71,18 @@ function getBidRequestData(bidReqConfig, callback, config) {
         deepSetValue(ortb2Site, 'ext.data.mobianGenres', genres);
         deepSetValue(ortb2Site, 'ext.data.apValues', apValues);
 
+        setKeyValue('mobianRisk', mobianRisk);
```

Comment:
RTD modules can provide targeting via `getTargetingData` - https://docs.prebid.org/dev-docs/add-rtd-submodule.html#gettargetingdata

Would that work for your use case? I think doing it like this has issues - it doesn't necessarily happen at the right time, so you may be setting up targeting for the wrong slot or it may be cleared later by prebid's targeting logic.

Reply:
So what we need to set the targeting for every ad slot / unit using the googletag API as shown in my code.

How would you approach this using `getTargetingData`? You mean invoking `setKeyValue` from within the `getTargetingData` function?

Reply:
No - `getTargetingData` returns a map of maps (ad unit code -> key -> value), which is taken into account by prebid targeting APIs (`getAdserverTargeting`, `setTargetingForGPTAsync` & co), which in turn call the GPT targeting APIs. 

It's a way to integrate your targeting with Prebid - compared to what you have here, it would only be applied when and where the publisher asks for targeting, and they'd know how to configure it.

Reply:
We do not need to integrate into the bid, we just need to set the targeting on a page level (https://developers.google.com/publisher-tag/guides/key-value-targeting#set_targeting)

Browsi is already doing this:
https://github.com/prebid/Prebid.js/blob/master/modules/browsiRtdProvider.js#L70

Are they doing it wrong?

Reply:
it seems I was unaware of this prebid method to do it, whats the disadvantage of moving to it?

Reply:
we decided to greenlight your implementation, however, you should modify the browsi adapter as part of your pr to import the function and use the duplicated code you identified